splink/linker.py

```diff
@@ -469,6 +463,27 @@ def _execute_sql_against_backend(self, sql, templated_name, physical_name):
             f"_execute_sql_against_backend not implemented for {type(self)}"
         )
 
+    def _do_execute_sql(self, final_sql, templated_name, physical_name):
+        """Actually execute the sql against the backend database.
+
+        This could return something, or not. It's up to the Linker subclass to decide.
+        """
+        raise NotImplementedError(f"_do_execute_sql not implemented for {type(self)}")
+
+    def _log_and_do_execute_sql(self, final_sql, templated_name, physical_name):
+        """Log the sql, then call _do_execute_sql(), wrapping any errors"""
+        logger.debug(execute_sql_logging_message_info(templated_name, physical_name))
+        logger.log(5, log_sql(final_sql))
+        try:
+            return self._do_execute_sql(final_sql, templated_name, physical_name)
+        except Error as e:
```

Comment:
I can see this is being used elsewhere in the code, but I don't believe we want to use the `Error` class from copy (here and elsewhere too...).

See the [copy docs](https://docs.python.org/3/library/copy.html).

Reply:
Oh oops, definitely not. Autocomplete in my IDE suggested it and I just blindly accepted it ;)